frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -136,35 +143,93 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
+                {
+                  metadata: {
+                    name: subscription.status.installedCSV,
+                    namespace: subscription.metadata.namespace,
+                  },
                 },
-              },
-              {},
-              deleteOptions,
-            ),
-          ]
-        : []),
-      ...(removePlugins
-        ? [k8sPatch(ConsoleOperatorConfigModel, consoleOperatorConfig, [patch])]
-        : []),
-    ];
-    const operandDeletionPromises = deleteOperands
-      ? operands.map((operand: K8sResourceCommon) => {
-          const model = modelFor(referenceFor(operand));
-          return k8sKill(model, operand, {}, deleteOptions);
+                {},
+                deleteOptions,
+              ),
+            ]
+          : []),
+        ...(removePlugins
+          ? [k8sPatch(ConsoleOperatorConfigModel, consoleOperatorConfig, [patch])]
+          : []),
+      ];
+
+      handleOperatorUninstallPromise(Promise.all(operatorUninstallPromises))
+        .then(() => {
+          setOperatorUninstallFinished(true);
         })
-      : [];
-    if (operandDeletionPromises.length > 0) {
+        .catch(() => {
+          setOperatorUninstallFinished(true);
+        });
+    };
+
+    const setOperandErrorsTo = (
+      operators: K8sResourceCommon[],
+      errorMsg: string,
+    ): OperandError[] => {
+      return operators.reduce((acc: OperandError[], curr, i) => {
+        return acc.concat({
+          operand: operands[i],
+          errorMessage: errorMsg,
+        });
+      }, []);
+    };
+
+    const verifyDeletionOfOperands = () => {
+      const url = `${window.SERVER_FLAGS.basePath}api/list-operands?name=${subscriptionName}&namespace=${subscriptionNamespace}`;
+      let operandVerificationErrors: OperandError[] = [];
+      let intervalID: Timeout;
+
+      const finishVerification = () => {
+        clearInterval(intervalID);
+        setOperandsDeleteInProgress(false);
+        setOperandsDeleteFinished(true);
+        if (operandVerificationErrors.length === 0) {
+          uninstallOperator();
+        } else {
+          // show operand deletion verification errors on results page
+          setOperandDeletionErrors(operandVerificationErrors);
+          setOperatorUninstallFinished(true);
+        }
+      };
+
+      intervalID = setInterval(() => {
+        coFetchJSON(url)
+          .then((curOperands) => {
+            setNumberOperandsRemaining(curOperands.items.length);
+            if (curOperands.items.length === 0) {
+              setTimeout(() => finishVerification(), 1000); // allow '0 Operands remaining' to display for a second
+            }
+          })
+          .catch(() => {
+            operandVerificationErrors = setOperandErrorsTo(
+              operands,
+              t('olm~error listing operand to verify deletion'),
+            );
+            finishVerification();
+          });
+      }, 2000); // every 2 seconds
```

Comment:
Ideally we'd poll quickly to start, and then if we discover it's a long running operator, we throttle this back to a less frequent interval.

Reply:
I could look into implementing this if you feel it is needed.